src/deepsparse/pipeline.py

```diff
@@ -178,63 +179,73 @@ async def run_async(self, *args, inference_state: InferenceState, **kwargs):
 
         next_step = self.router.START_ROUTE
         operator_output = None
+        if (
+            not hasattr(inference_state, "timer")
+            or getattr(inference_state, "timer") is None
+        ):
+            timer = self.timer_manager.get_new_timer()
+            inference_state.set_timer(timer)
 
-        while next_step != self.router.END_ROUTE:
-            # Check if running streaming; if that is the case, will return
-            # an AsyncGenerator. This requires the pipeline to support
-            # streaming with a generator_router set
-            if inference_state.current_state.get("streaming"):
-                return self._run_generate_async(
-                    operator_output=operator_output,
-                    inference_state=inference_state,
-                    next_step=next_step,
-                )
-
-            # Non Streaming/Generator pathway
-            if next_step == self.router.SPLIT_ROUTE:
-                if operator_output is None:
-                    raise ValueError(
-                        f"{self.router.SPLIT_ROUTE} should appear after "
-                        f"{self.ROUTER.START_ROUTE}"
-                    )
-
-                operator_output = await self._apply_split_async(
-                    operator_output, inference_state, loop=loop
-                )
-                next_step = self.router.JOIN_ROUTE
-
-            else:
-                if next_step == self.router.START_ROUTE:
-                    outputs = self.run_func(
-                        *args,
-                        func=self._scheduler_group.submit,
-                        operator=self.ops[next_step],
+        with inference_state.time(id="total"):
```

Comment:
would it be possible for us to be consistent with the existing timer? I believe we store `total_inference` with that one but you'd have to double check. Otherwise, the benchmark pipeline will have a lot of custom code for v2.

Reply:
THis is from the design doc, feel free to change if youd like. shouldnt be blocking merge

Reply:
We'll have to make it clear then that this isn't compatible with the `benchmark_pipeline`

Reply:
`total_inference` is fine